catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,25 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDatasIfNeeded(mode);
+	}
+
+	private void rotateLookDatasIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(true);
+		}
+	}
+
+	public void flipAllLookDatas() {
+		for (LookData lookData : sprite.getLookDataList()) {
```

Comment:
I tested it, you are right, on the first time when the need to flip occurs the UI Thread blocks for a very short but noticeable amount of time. I didn't know about `TextureRegion.isFlipX` which seems like a really good thing to do. I will look into that.